nmoscommon/webapi.py

```diff
@@ -571,112 +617,127 @@ def inner(*args, **kwargs):
-                                methods=methods + ["OPTIONS",],
-                                endpoint=f.__name__)(f)
-                    elif hasattr(value, "sockets_on"):
-                        socket_recv_gen = getattr(routesObject, "on_websocket_connect", None)
-                        if socket_recv_gen is None:
-                            f = self.handle_sock(expects_json(value), self.socks)
+                            basepath + routesFunction.app_resource_route + '<path:path>/',
+                            methods=methods + ["OPTIONS", ],
+                            endpoint=f.__name__)(f)
+
+                    elif hasattr(routesFunction, "socket_path"):
+                        websocket_opened = getattr(routesClass, "on_websocket_connect", None)
+                        if websocket_opened is None:
+                            self.sockets.route(
+                                basepath + routesFunction.socket_path,
+                                endpoint=endpoint)(
+                                    self.handle_sock(
+                                        expects_json(routesFunction), self.socks))
                         else:
-                            f = socket_recv_gen(expects_json(value))
-                        self.sockets.route(basepath + value.sockets_on, endpoint=endpoint)(f)
-                    elif hasattr(value, "errorhandler_args"):
-                        if value.errorhandler_args:
+                            self.sockets.route(
+                                basepath + routesFunction.socket_path,
+                                endpoint=endpoint)(
+                                    websocket_opened(
+                                        expects_json(routesFunction)))
+
+                    elif hasattr(routesFunction, "errorhandler_args"):
+                        if routesFunction.errorhandler_args:
                             self.app.errorhandler(
-                                *value.errorhandler_args, **
-                                value.errorhandler_kwargs)(crossdomain(
-                                    origin='*',
-                                    methods=["GET", "POST", "PUT", "DELETE",
-                                             "OPTIONS", "HEAD"])(dummy(value)))
+                                *routesFunction.errorhandler_args,
+                                **routesFunction.errorhandler_kwargs)(
+                                    crossdomain(
+                                        origin='*',
+                                        methods=["GET", "POST", "PUT", "DELETE", "OPTIONS", "HEAD"])(
+                                            dummy(routesFunction)))
                         else:
                             for n in range(400, 600):
-                                try:
+                                try:  # Apply errorhandlers for all known error cdes
```

Comment:
Minor typo at the end
```suggestion
                                try:  # Apply errorhandlers for all known error codes
```